src/app/services/account.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { BehaviorSubject, first,Subject,tap } from 'rxjs';
+import { User } from '../Users';
+import { Router } from '@angular/router';
+
+@Injectable({
+  providedIn: 'root'
+})
+ export class AccountService {
+  public redirectUrl?: string; 
+  private currentUserSubject = new BehaviorSubject<User | string>("default"); 
+  private localStorageUserSubject = new BehaviorSubject<User | string | null>("default");
+  constructor(private http: HttpClient, private router: Router) {  
+  }
+
+  getCurrentUserSubject(): BehaviorSubject<User | string> {
+    return this.currentUserSubject; 
+  }
+  getlocalStorageUserSubject(): BehaviorSubject<any> {
+   return this.localStorageUserSubject; 
+  }
```

Comment:
Consider a Subject as a private observable. You don't want to share that Subject because then any calling code could write to your observable and change your state or create errors. If you want to expose the result of your subject you should create an observable using `.asObservable()` on the subject and make that Observable public.

Reply:
The tutorial I was following did not mention this yet, thank you for the heads up.
I am going to continue to read up on the best practices of RxJs outside of my tutorial here: https://betterprogramming.pub/rxjs-best-practices-7f559d811514